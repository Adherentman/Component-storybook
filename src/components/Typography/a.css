superText {
  font-family: "lucida grande lucida sans unicode lucida helvetica Hiragino Sans GB Microsoft YaHei WenQuanYi Micro Hei sans-serif";
  font-size: 88px;
  font-weight: 600;
  line-height: 1px;
}