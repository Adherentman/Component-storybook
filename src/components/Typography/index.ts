import TextSuper from './textSuper';

export {
  TextSuper
}