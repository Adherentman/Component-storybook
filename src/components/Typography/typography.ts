import {
  fontSizeSuper,
  fontWeightMedium,
  lineHeightSuper
} from "../../styles/typography";
import font from '../../utils/font';

export function textSuper() {
  return {
    fontFamily: font(),
    fontSize: `${fontSizeSuper}px`,
    fontWeight: fontWeightMedium,
    lineHeight: lineHeightSuper,
  };
}