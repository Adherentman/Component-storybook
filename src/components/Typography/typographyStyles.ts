import {
	fontSizeHeading1,
	fontSizeHeading2,
	fontSizeHeading3,
	fontSizeHeading4,
	fontSizeHeading5,
	fontSizeHeading6,
	fontSizeHeading7,
	fontSizeHeading8,
	fontSizeSuper,
	fontWeightMedium,
	fontWeightRegular,
	lineHeightHeading1,
	lineHeightHeading2,
	lineHeightHeading3,
	lineHeightHeading4,
	lineHeightHeading5,
	lineHeightHeading6,
	lineHeightHeading7,
	lineHeightHeading8,
	lineHeightSuper
} from "../../styles/typography";
import font from '../../utils/font';

export interface ItypographyStylesType {
	root: any,
	superStyle: any,
	sizeHeading1: any,
	sizeHeading2: any,
	sizeHeading3: any,
	sizeHeading4: any,
	sizeHeading5: any,
	sizeHeading6: any,
	sizeHeading7: any,
	sizeHeading8: any
}

export const typographyStyles: ItypographyStylesType = {
	root: {
		fontFamily: font(),
	},
	superStyle: {
		fontSize: fontSizeSuper,
		fontWeight: fontWeightMedium,
		lineHeight: lineHeightSuper,
	},
	// tslint:disable-next-line:object-literal-sort-keys
	sizeHeading1: {
		fontSize: fontSizeHeading1,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading1,
	},
	sizeHeading2: {
		fontSize: fontSizeHeading2,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading2,
	},
	sizeHeading3: {
		fontSize: fontSizeHeading3,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading3,
	},
	sizeHeading4: {
		fontSize: fontSizeHeading4,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading4,
	},
	sizeHeading5: {
		fontSize: fontSizeHeading5,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading5,
	},
	sizeHeading6: {
		fontSize: fontSizeHeading6,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading6
	},
	sizeHeading7: {
		fontSize: fontSizeHeading7,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading7
	},
	sizeHeading8: {
		fontSize: fontSizeHeading8,
		fontWeight: fontWeightRegular,
		lineHeight: lineHeightHeading8
	}
};
 