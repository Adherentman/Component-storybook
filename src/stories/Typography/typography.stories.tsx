import { withInfo } from "@storybook/addon-info";
import { select, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as PropTypes from "prop-types";
import * as React from "react";
// import JSXAddon from "storybook-addon-jsx";
import { Typography } from "../../components";
import TypographyStory from "./typography";

storiesOf("Typography", module)
  .addDecorator(withKnobs)
  .add("Typography", () => <TypographyStory />)
  .add(
    "super —— H8",
    withInfo({
      propTables: [Typography],
      text: "哒哒哒哒哒"
    })(() => (
      <Typography
        size={select(
          "Size",
          [
            "super",
            "H1",
            "H2",
            "H3",
            "H4",
            "H5",
            "H6",
            "H7",
            "H8",
            "article",
            "articleSmall",
            "bodySmall",
            "accent",
            "uppercase"
          ],
          "super"
        )}
      >
        Amazing Typography
      </Typography>
    ))
  );

Typography.displayName = "Typography";

Typography.defaultProps = {
  size: "root"
};

Typography.propTypes = {
  /**
   * @description big size
   */
  size: PropTypes.oneOf([
    "super",
    "H1",
    "H2",
    "H3",
    "H4",
    "H5",
    "H6",
    "H7",
    "H8",
    "article",
    "articleSmall",
    "bodySmall",
    "accent",
    "uppercase"
  ])
};
