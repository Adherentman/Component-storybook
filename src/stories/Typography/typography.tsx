import * as React from "react";
import { Typography } from "../../components";
// import styles from '../styles.css';

const data = [
  { size: "super", value: "Super Typography" },
  { size: "H1", value: "H1 Typography" },
  { size: "H2", value: "H2 Typography" },
  { size: "H3", value: "H3 Typography" },
  { size: "H4", value: "H4 Typography" },
  { size: "H5", value: "H5 Typography" },
  { size: "H6", value: "H6 Typography" },
  { size: "article", value: "article Typography" },
  { size: "articleSmall", value: "articleSmall Typography" },
  { size: "bodySmall", value: "bodySmall Typography" },
  { size: "accent", value: "accent Typography" },
  { size: "uppercase", value: "uppercase Typography" }
];

const styles = {
  main: {
    fontFamily: "Helvetica Neue', 'Helvetica', 'Segoe UI', 'Arial', sans-serif",
    lineHeigh: 1.4,
    maxWidth: 1290,
    // tslint:disable-next-line:object-literal-sort-keys
    margin: 15
  },
  smallText: {
    color: "rgba(0, 0, 0, .3)",
    fontSize: "16px",
    lineHeight: 3,
    marginBottom: "1em"
  }
};
class TypographyStory extends React.Component {
  public render() {
    return (
      <div style={styles.main}>
        <h1>Typography</h1>
        {data.map((x, index) => (
          <React.Fragment key={index}>
            <div>
              <small style={styles.smallText}>{x.size}</small>
            </div>
            <Typography size={x.size}>{x.value}</Typography>
          </React.Fragment>
        ))}
      </div>
    );
  }
}

export default TypographyStory;
