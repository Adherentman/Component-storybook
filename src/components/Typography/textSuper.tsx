import * as React from 'react';
import { textSuper } from '..//Typography/typography';

function TextSuper({children}: any) {
  return <React.Fragment>
    <div style={textSuper()}>
      {children}
    </div>
  </React.Fragment>
}

export default TextSuper;