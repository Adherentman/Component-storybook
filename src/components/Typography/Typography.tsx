import * as classNames from "classnames";
import * as React from 'react';
import injectSheet from 'react-jss';
import typographyStyles from '../Typography/typographyStyles';

function Typography({children, ...props}: any) {

  const { size, classes } = props;

  const typographyStyle = classNames(
    classes.root,
    {
      [classes.superStyle]: size === 'superText'
    }
)

  return <React.Fragment>
    <div className={typographyStyle}>
      <span>{children}</span>
    </div>
  </React.Fragment>
}

export default injectSheet(typographyStyles)(Typography);