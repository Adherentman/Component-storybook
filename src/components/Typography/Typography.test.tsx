import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Typography from './index';

test('Typography change className', () => {
  const components = renderer.create(
    <Typography>是是是</Typography>
  ).toJSON();
  expect(components).toMatchSnapshot();
})