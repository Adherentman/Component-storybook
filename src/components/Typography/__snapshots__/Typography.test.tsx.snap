// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography change className 1`] = `
<span
  className="Typography-root-0-1-1"
>
  是是是
</span>
`;
